pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -857,22 +857,24 @@ public static void deactivateKitRequest(int dsmKitRequestId, @NonNull String dea
                     results.resultException);
         }
         if (Objects.nonNull(ddpInstanceDto)) {
-            KitRequestShipping kitRequestShipping = new KitRequestShipping();
-            kitRequestShipping.setDsmKitRequestId(dsmKitRequestId);
-            kitRequestShipping.setDeactivationReason(deactivationReason);
-            kitRequestShipping.setDeactivatedDate(deactivatedDate);
-            kitRequestShipping.setKitTypeName(kitRequestDao.getKitTypeByKitRequestId(dsmKitRequestId));
-            try {
-                //todo: if multiple kits exist for same kit-request(dsmKitRequestId), we are updating all elements? edge case though
-                UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping, ddpInstanceDto,
-                        ESObjectConstants.DSM_KIT_REQUEST_ID, ESObjectConstants.DSM_KIT_REQUEST_ID, dsmKitRequestId,
-                        new PutToNestedScriptBuilder()).export();
-            } catch (Exception e) {
-                logger.error(String.format("Error updating kit request shipping deactivate reason with dsm kit request id: %s in "
-                        + "ElasticSearch", dsmKitRequestId));
-                e.printStackTrace();
+            if (ddpInstanceDto.isESUpdatePossible()) {
+                //skip if no elastic index. ex: juniper studies/eating-disorder
+                KitRequestShipping kitRequestShipping = new KitRequestShipping();
+                kitRequestShipping.setDsmKitRequestId(dsmKitRequestId);
+                kitRequestShipping.setDeactivationReason(deactivationReason);
+                kitRequestShipping.setDeactivatedDate(deactivatedDate);
+                kitRequestShipping.setKitTypeName(kitRequestDao.getKitTypeByKitRequestId(dsmKitRequestId));
+                try {
+                    //todo: if multiple kits exist for same kit-request(dsmKitRequestId), we are updating all elements? edge case though
+                    UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping, ddpInstanceDto,
+                            ESObjectConstants.DSM_KIT_REQUEST_ID, ESObjectConstants.DSM_KIT_REQUEST_ID, dsmKitRequestId,
+                            new PutToNestedScriptBuilder()).export();
+                } catch (Exception e) {
+                    logger.error(String.format("Error updating kit request shipping deactivate reason with dsm kit request id: %s in "
+                            + "ElasticSearch", dsmKitRequestId));
+                    e.printStackTrace();
+                }
             }
-
         } else {
```

Comment:
wondered if we ever get to this else ? 
when would ddpInstanceDto be null? All realms in DSM should have a ddpInstanceDto !

Reply:
I agree.
Looking at the callers, there is only one possibility of ddpInstanceDto being null, and that seems impossible/unlikely.
I suggest putting a 'logger.error' statement here, along with a comment that says we should remove the code in the future if the error never reports.
That at least provides an opinion about the code and protection in the unlikely case that the code path is used.

Does that make sense?

Reply:
Adding error or some log message will tell us if it ever gets there but I was more of wondering when we should try to do refund (below else code) . As it stands its probably never invoked and probably we're not requesting refunds. Wanted to check with Whitney, if there is a refund scenario at all in deactivate. If so it probably should be invoked for all deactivate and has a label assuming we get charged once label is created.
if (easypostApiKey != null) {
       KitRequestShipping.refundKit(dsmKitRequestId, easypostApiKey, ddpInstanceDto);
}

I will add a error log though